### PR TITLE
Improve speed of GetPointsInRadius

### PR DESCRIPTION
**Pull request by Flamefire**
- Preallocate memory (test map load time decreased from 18s to 14s in debug mode)
- Auto-deduce result_type and remove that typedef (now allows lambdas)

### CODE REVIEW

File: libs/s25main/figures/nofScout_Free.cpp [hunk 1]
```diff
@@ -156,7 +156,7 @@ struct IsScoutable
 void nofScout_Free::GoToNewNode()
 {
     std::vector<MapPoint> available_points =
-      gwg->GetPointsInRadius<-1>(flag->GetPos(), SCOUT_RANGE, Identity<MapPoint>(), IsScoutable(player, *gwg));
+      gwg->GetMatchingPointsInRadius<-1>(flag->GetPos(), SCOUT_RANGE, IsScoutable(player, *gwg));
```
Comment (Flow86) [inline libs/s25main/figures/nofScout_Free.cpp:159]: no test for this?

File: libs/s25main/figures/nofWorkman.cpp [hunk 2]
```diff
@@ -119,8 +119,8 @@ struct NodeHasResource
 MapPoint nofWorkman::FindPointWithResource(ResourceType type) const
 {
     // Alle Punkte durchgehen, bis man einen findet, wo man graben kann
-    std::vector<MapPoint> pts =
-      gwg->GetPointsInRadius<1>(pos, MINER_RADIUS, Identity<MapPoint>(), NodeHasResource(*gwg, type), true);
+    const std::vector<MapPoint> pts =
```
Comment (Flow86) [inline libs/s25main/figures/nofWorkman.cpp:122]: no test for this?

File: libs/s25main/figures/nofScout_Free.cpp (hunk 1 above)
Comment (Flamefire) [reply libs/s25main/figures/nofScout_Free.cpp:159]: No not yet. And IMO out of scope for this PR. Adding tests for the figures/jobs is quite complicated as you need to set up a whole world etc. Might do later but not in this PR. I think it is clear from the change that it doesn't change behavior if the function works as expected and that at least is tested

File: libs/s25main/figures/nofWorkman.cpp (hunk 2 above)
Comment (Flamefire) [reply libs/s25main/figures/nofWorkman.cpp:122]: see above

**Review by Flow86 — APPROVED** [review]

File: libs/s25main/figures/nofScout_Free.cpp (hunk 1 above)
Comment (Flamefire) [reply libs/s25main/figures/nofScout_Free.cpp:151]: Test added

File: libs/s25main/figures/nofWorkman.cpp (hunk 2 above)
Comment (Flamefire) [reply libs/s25main/figures/nofWorkman.cpp:122]: Test added

**Pull request merged**